tests/cli/test_cmds_local_run.py

```diff
@@ -1963,6 +1989,102 @@ def test_missing_volumes_skipped(mock_exists):
         assert kwargs["volumes"] == []
 
 
+@mock.patch(
+    "paasta_tools.cli.cmds.local_run.pick_random_port",
+    autospec=True,
+    return_value=666,
+)
+@mock.patch("paasta_tools.cli.cmds.local_run.get_docker_run_cmd", autospec=True)
+@mock.patch("paasta_tools.cli.cmds.local_run.execlpe", autospec=True)
+@mock.patch(
+    "paasta_tools.cli.cmds.local_run.decrypt_secret_volumes",
+    autospec=True,
+    return_value={},
+)
+@mock.patch(
+    "paasta_tools.cli.cmds.local_run._run",
+    autospec=True,
+    return_value=(0, "fake _run output"),
+)
+@mock.patch("paasta_tools.cli.cmds.local_run.get_container_id", autospec=True)
+@mock.patch(
+    "paasta_tools.cli.cmds.local_run.get_healthcheck_for_instance",
+    autospec=True,
+    return_value=("fake_healthcheck_mode", "fake_healthcheck_uri"),
+)
+@mock.patch(
+    "paasta_tools.cli.cmds.local_run.open",
+    new_callable=mock.mock_open(),
+    autospec=None,
+)
+@mock.patch("os.makedirs", autospec=True)
+@mock.patch(
+    "paasta_tools.cli.cmds.local_run.assume_aws_role",
+    autospec=True,
+    return_value={"access_key": "abcdefg", "secret_key": "abcdefg"},
+)
```

Comment:
using the context manager form of mocks is a little more preferable than the decorator version (easier to scope the changes to just the code that needs the mocks active and easier to update the mocks used without needing to also update the function signature) - but we use this form enough already that it's likely not  worth updating this unless this comment somehow speaks to you :p

Reply:
True, mostly just kept with the style of the existing tests.